### PR TITLE
hal/ia32: Add HPET

### PR DESCRIPTION
**Pull request by astalke**
JIRA: RTOS-530

## Description

Change in plo that is required for https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/476 to work.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: ia32-generic-pc, ia32-generic-qemu

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/476
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: hal/ia32/acpi.c [hunk 1]
```diff
@@ -79,12 +79,23 @@ static void _hal_acpiFadtHandler(hal_syspage_t *hs, sdt_header_t *header)
...
+
 static const struct {
 	const char magic[4];
 	void (*handler)(hal_syspage_t *hs, sdt_header_t *);
 } acpi_knownTables[] = {
-	{ .magic = "APIC", .handler = _hal_acpiMadtHandler },
-	{ .magic = "FACP", .handler = _hal_acpiFadtHandler }
+	  { .magic = "APIC", .handler = _hal_acpiMadtHandler }
+	, { .magic = "FACP", .handler = _hal_acpiFadtHandler }
+	, { .magic = "HPET", .handler = _hal_acpiHpetHandler }
```
Comment (agkaminski) [inline hal/ia32/acpi.c:98]:
```suggestion
	{ .magic = "APIC", .handler = _hal_acpiMadtHandler },
	{ .magic = "FACP", .handler = _hal_acpiFadtHandler },
	{ .magic = "HPET", .handler = _hal_acpiHpetHandler }
```

File: hal/ia32/acpi.c [hunk 2]
```diff
@@ -79,12 +79,23 @@ static void _hal_acpiFadtHandler(hal_syspage_t *hs, sdt_header_t *header)
 	}
 }
 
+
+static void _hal_acpiHpetHandler(hal_syspage_t *hs, sdt_header_t *header)
+{
+	if (_hal_checksumVerify(header, header->length) != 0) {
+		hs->hpet = (unsigned int)header;
```
Comment (agkaminski) [inline hal/ia32/acpi.c:86]: `hs->hpet` should be of `addr_t` type

Comment (agkaminski) [reply hal/ia32/acpi.c:86]: Ok, at least go with `unsigned long` + `FIXME` for now

Comment (astalke) [reply hal/ia32/acpi.c:86]: Why `unsigned long`? `addr_t` is defined as `u32` which is defined as `unsigned int`

Comment (agkaminski) [reply hal/ia32/acpi.c:86]: In case we try to make a x64 port in the meantime

Comment (agkaminski) [reply hal/ia32/acpi.c]: Perphaps `FIXME - should be addr_t`?

**Review by agkaminski — APPROVED** [review]
LGTM

**Pull request merged**